app/forms/ReceiveGoodsNIPForm.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.FormValidation._
+import models.ConditionalValue
+import play.api.data.Form
+import play.api.data.Forms._
+import uk.gov.voa.play.form.ConditionalMappings._
+
+object ReceiveGoodsNIPForm extends RequiredBooleanForm {
+  override val errorMsg = "validation.northernIrelandReceiveGoods.missing"
+  val INPUT_NORTHERN_IRELAND_AMOUNT: String = "northernIrelandReceiveGoods"
+  implicit val errorCode: ErrorCode = INPUT_NORTHERN_IRELAND_AMOUNT
+  val RADIO_YES_NO: String = "value"
+
+  val form = Form(
+    tuple(
+      RADIO_YES_NO -> requiredBoolean,
+      INPUT_NORTHERN_IRELAND_AMOUNT -> mandatoryIf(
+        isEqual(RADIO_YES_NO, "true"),
+        text.verifying(mandatoryFullNumericText)
+          .transform[BigDecimal](string =>
+            BigDecimal(string.replace(",", "")).setScale(2, BigDecimal.RoundingMode.HALF_UP),
+            _.toString
+          )
+      )
+    )
+  )
+
+
+  def fillWithPrePop(optNorthernIrelandReceiveGoodsFormData: Option[ConditionalValue]): Form[(Boolean, Option[BigDecimal])] = {
```

Comment:
Just do this in the show in the controller, no point to split it out. In fact there's only one place where we seem to do that so it's really not our pattern.